pkg/schema/schema.go

```diff
@@ -143,10 +158,63 @@ func newCollectionItemValueType(collectionItemValue interface{}) (yamlmeta.Type,
 	return nil, fmt.Errorf("Collection item type did not match any known types")
 }
 
+func DefaultDataValues(doc *yamlmeta.Document) (*yamlmeta.Document, error) {
+	docCopy := doc.DeepCopyAsNode()
+	for _, value := range docCopy.GetValues() {
+		setDefaultValues(value)
+	}
+
+	return docCopy.(*yamlmeta.Document), nil
+}
+
+func setDefaultValues(node interface{}) {
+	switch typedNode := node.(type) {
+	case *yamlmeta.Map:
+		for _, value := range typedNode.Items {
+			setDefaultValues(value)
+		}
+	case *yamlmeta.MapItem:
+		setDefaultValues(typedNode.Value)
+	case *yamlmeta.Array:
+		typedNode.Items = []*yamlmeta.ArrayItem{}
+	}
+}
+
 func (as *AnySchema) AssignType(typeable yamlmeta.Typeable) yamlmeta.TypeCheck {
 	return yamlmeta.TypeCheck{}
 }
 
+func (n NullSchema) AssignType(typeable yamlmeta.Typeable) yamlmeta.TypeCheck {
+	return yamlmeta.TypeCheck{}
+}
+
 func (s *DocumentSchema) AssignType(typeable yamlmeta.Typeable) yamlmeta.TypeCheck {
 	return s.Allowed.AssignTypeTo(typeable)
 }
+
+func (as *AnySchema) AsDataValue() *yamlmeta.Document {
+	return nil
+}
+
+func (n NullSchema) AsDataValue() *yamlmeta.Document {
+	return nil
+}
+
+func (s *DocumentSchema) AsDataValue() *yamlmeta.Document {
+	return s.defaultDVs
+}
+
+func (as *AnySchema) ValidateWithValues(valuesFilesCount int) error {
+	return nil
+}
+
+func (n NullSchema) ValidateWithValues(valuesFilesCount int) error {
```

Comment:
I am a little bit confused about this function. Is the intent of the function to error out if a schema is NOT provided and no data values files are present?
Not sure if Schema should be responsible for this check. Maybe the LibraryLoader should have `#HasSchemaFiles() bool` and `#HasValuesFiles() bool`. This way at the `cmd` level we can check if the above is conditions are met to continue or not.

Reply:
> I am a little bit confused about this function. Is the intent of the function to error out if a schema is NOT provided and no data values files are present?
No, the intent is for when schema is enabled, no schema is provided, and there are data values. Then it should error out.

It is a good observation. We think that the current implementation is sufficient for the moment. We should discuss this further. We've added it to tech forum.